doc/source/development/rfc/rfc94_field_precision_width_metadata.rst

```diff
@@ -0,0 +1,136 @@
+.. _rfc-94:
+
+=============================================================
+RFC 94: Numeric fields width/precision metadata
+=============================================================
+
+============== =============================================
+Author:        Alessandro Pasotti
+Contact:       elpaso @ itopen.it
+Started:       2023-Feb-17
+Status:        Development
+Target:        GDAL 3.7
+============== =============================================
+
+Summary
+-------
+
+The document proposes and describes the introduction of a new method
```

Comment:
For creation use cases, where on wants to convert from one convention to another one, one needs to know in advance the convention that will be used by the output driver.
Perhaps have a GDAL_DMD_NUMERIC_FIELD_WITH_INCLUDES_DECIMAL_SEPARATOR driver metadata item whose value could be "YES" or "NO" (and assumed to be unknown if not set)
Similarly GDAL_DMD_NUMERIC_FIELD_WITH_INCLUDES_SIGN

Reply:
I agree that that information shouldn't change within a single layer but since I do not know enough about all the involved drivers I preferred to stay on the safe side and make it a field property. 

Anyway, this can be decided at a later stage of the analysis when we are sure there aren't different field behaviors within a single layer.

Reply:
my point in having that at a driver level was for a converter application (ogr2gr, QGIS vector file writer, etc.), that needs to create numeric fields. It needs to know *before* creating them what the convention will be to be able to add +1 or +2 to the width if needed.

Reply:
I've changed the proposed implementation to a driver metadata entry.